ecs-cli/modules/cli/attributechecker/attribute_checker_app.go

```diff
@@ -0,0 +1,169 @@
+	}
+	commandConfig, err := config.NewCommandConfig(c, rdwr)
+	if err != nil {
+		logrus.Fatal("Error executing 'Attribute Checker': ", err)
+	}
+
+	ecsClient := ecsclient.NewECSClient(commandConfig)
+	//ecsClient.Initialize(commandConfig)
+
+	taskDefAttributeNames, err := taskdefattributesCheckRequest(c, ecsClient)
+	if err != nil {
+		logrus.Fatal("Error executing 'Attribute Checker': ", err)
+	}
+	if len(taskDefAttributeNames) == 0 {
+		fmt.Println("Valid Error: Task Definition has no attributes")
+	}
+
+	descrContainerInstancesResponse, err := describeContainerInstancesAttributeMap(c, ecsClient, commandConfig)
+	if err != nil {
+		logrus.Fatal("Error executing 'Attribute Checker': ", err)
+	}
+
+	compareOutput := compare(taskDefAttributeNames, descrContainerInstancesResponse)
+	result := ConvertToInfoSet(compareOutput)
+	os.Stdout.WriteString(result.String(infoColumns, displayTitle))
+}
+
+//compares between container instances and Task definition. Prints missing attribute names
+func compare(taskDefAttributeNames []*string, descrContainerInstancesResponse map[string][]*string) map[string]string {
+	attributeCheckerResult := make(map[string]string)
+	for containerInstanceARN, containerInstanceAttriubeNames := range descrContainerInstancesResponse {
+		var taskdefAttributeNames []string
+		for i := 0; i < len(taskDefAttributeNames); i++ {
+			for j := 0; j < len(containerInstanceAttriubeNames); j++ {
+				if *taskDefAttributeNames[i] != *containerInstanceAttriubeNames[j] {
+					if j == len(containerInstanceAttriubeNames)-1 {
+						taskdefAttributeNames = append(taskdefAttributeNames, *taskDefAttributeNames[i])
+					}
+					continue
+				}
+				break
+			}
+		}
+		missingAttributes := strings.Join(taskdefAttributeNames, ", ")
+		if len(missingAttributes) == 0 {
+			missingAttributes = "None"
+		}
+		containerInstance := strings.Split(containerInstanceARN, "/")
+		attributeCheckerResult[containerInstance[1]] = missingAttributes
+	}
+	return attributeCheckerResult
+}
+
+// DescribeContainerInstancesAttributeMap and get a map with Container instance ARN and Container instances attribute Names
+func describeContainerInstancesAttributeMap(context *cli.Context, ecsClient ecsclient.ECSClient, commandConfig *config.CommandConfig) (map[string][]*string, error) {
+	if err := validateCluster(commandConfig.Cluster, ecsClient); err != nil {
+		return nil, err
+	}
+	var containerInstanceIdentifiers []*string
+	containerInstanceIdentifier := context.String(flags.ContainerInstancesFlag)
+	splitValues := strings.Split(containerInstanceIdentifier, ",")
+	containerInstanceIdentifiers = aws.StringSlice(splitValues)
+
+	descrContainerInstancesAttributes, err := ecsClient.GetAttributesFromDescribeContainerInstances(containerInstanceIdentifiers)
+	if err != nil {
+		return nil, errors.Wrapf(err, fmt.Sprintf("Failed to Describe Container Instances; try using %s %s to Describe Containers Instances", flags.ContainerInstancesFlag, flags.ClusterFlag))
```

Comment:
The "try using %s %s to Describe Containers Instances" is probably unnecessary.

Reply:
Esp since you check in `validateAttributeCheckerFlags` that the container instance flag is present